scripts/browserstack.py

```diff
@@ -279,6 +250,51 @@ def executeTests(appUrl, testUrl, isNightly, testClasses):
         )
         return None
 
+def executeTests(appUrl, testUrl, isNightly, testClasses):
+    shards = testShards(isNightly, testClasses)
+    devices = []
+    if isNightly:
+        devices = [
+            "Google Pixel 7-13.0",
+            "Samsung Galaxy S22-12.0",
+        ]
+    else:
+        devices = [
+            "Samsung Galaxy S22-12.0",
+        ]
+    json = {
+        "app": appUrl,
+        "devices": devices,
+        "testSuite": testUrl,
+        "networkLogs": True,
+        "deviceLogs": True,
+        "video": True,
+        "acceptInsecureCerts": True,
+        "locale": "en_US",
+        "enableSpoonFramework": False,
+        "project": PROJECT_NAME,
+        "shards": {
+            "numberOfShards": len(shards),
+            "mapping": shards,
+        },
+    }
+    return executeTestsWithJson(appUrl, testUrl, json)
+
+def executeTestsForFailure(appUrl, testUrl, device, testClasses):
+    json = {
```

Comment:
instead of repeating the whole json in each function, you could pass a map with just the parameters that differ to `executeTestsWithJson` (maybe named something else). This would be useful to make sure that the other parameters stay aligned and so that it's obvious what the difference is between how we run tests on the first run vs. on a retry